File: network/src/main/java/com/lengjiye/network/SignInterceptor.java
package com.lengjiye.network;


import com.lengjiye.base.application.MasterApplication;
import com.lengjiye.tools.LogTool;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.MediaType;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.ResponseBody;

public class SignInterceptor implements Interceptor {

    @Override
    public Response intercept(Chain chain) throws IOException {

        Request originRequest = chain.request();

        Request.Builder requestBuilder = originRequest.newBuilder();
        // 可以添加公共请求参数
        Request request = requestBuilder.build();
        Response response = chain.proceed(request);
        ResponseBody body = response.body();
        if (body != null) {
            MediaType mediaType = body.contentType();
            LogTool.d("mediaType=" + mediaType);
            String content = body.string();
            if ("debug".equals(MasterApplication.getInstance().buildType())) {
                LogTool.e("#############################################################");
                LogTool.e("request.url()=" + request.url());
                LogTool.e("response.body()=" + content);
                LogTool.e("#############################################################");
            }
            return response.newBuilder().body(ResponseBody.create(mediaType, content)).build();
        }
        return response;
    }
}

File: network/src/main/java/com/lengjiye/network/interceptor/SignInterceptor.kt
package com.lengjiye.network.interceptor

import com.lengjiye.code.baseparameter.application.MasterApplication
import com.lengjiye.tools.log.logD
import com.lengjiye.tools.log.logE
import okhttp3.Interceptor
import okhttp3.Response
import okhttp3.ResponseBody
import java.io.IOException

class SignInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val originRequest = chain.request()
        val requestBuilder = originRequest.newBuilder()
        // 可以添加公共请求参数
        val request = requestBuilder.build()
        val response = chain.proceed(request)
        val body = response.body()
        if (body != null) {
            val mediaType = body.contentType()
            logD("mediaType=$mediaType")
            val content = body.string()
            if ("debug" == MasterApplication.getInstance().buildType()) {
                logE("#############################################################")
                logE("request.url()=" + request.url())
                logE("response.body()=$content")
                logE("#############################################################")
            }
            return response.newBuilder().body(ResponseBody.create(mediaType, content)).build()
        }
        return response
    }
}